Documentation packages are not handled correctly

```
FAILED tests/ert/unit_tests/docs/test_ert_documentation.py::test_that_forward_model_step_documentation_defaults_to_plugins_source_package[Stea] - AssertionError: abc.Stea documentation reports source_package=everest_models, expected abc
assert 'everest_models' == 'abc'
  
  - abc
  + everest_models
```
and
```
E           AssertionError: fmu.sim2seis.forward_models.sim2seis_cleanup.Cleanup documentation reports source_package=fmu.sim2seis, expected fmu
E           assert 'fmu.sim2seis' == 'fmu'
E             
E             - fmu
E             + fmu.sim2seis
```

This PR seems to introduce the breaking changes:
https://github.com/equinor/ert/pull/12670

ref.
https://github.com/equinor/komodo-releases/actions/runs/21353414052/job/61457681606

--- a/tests/ert/unit_tests/docs/test_ert_documentation.py
+++ b/tests/ert/unit_tests/docs/test_ert_documentation.py
@@ -121,20 +121,20 @@ def test_divide_into_categories_job_source(test_input, expected_source_package):
 
 
 @pytest.mark.parametrize("fm_step", ErtPluginManager().forward_model_steps)
-def test_that_forward_model_step_documentation_defaults_to_plugins_source_package(
+def test_that_forward_model_step_documentation_source_package_matches_the_actual_source_package(  # noqa: E501
     fm_step,
 ):
     # The test assumes that repr(fm_step) returns a string on format:
     # "<class 'fully.qualified.class.name'>"
     match_qualified_class_name = re.match(r"\<class '(.*?)'>", repr(fm_step))
     assert match_qualified_class_name, "Could not get qualified class name from fm_step"
-    qualified_class_name = match_qualified_class_name[1]
+    qualified_class_name: str = match_qualified_class_name[1]
 
-    source_package = qualified_class_name.split(".")[0]
     documentation = fm_step.documentation()
 
     if documentation:
-        assert documentation.source_package == source_package, (
-            f"{qualified_class_name} documentation reports source_package="
-            f"{documentation.source_package}, expected {source_package}"
+        assert qualified_class_name.startswith(documentation.source_package), (
+            f"Documentation mismatch in {fm_step!r} "
+            f"source_package ('{documentation.source_package}') does not match "
+            f"the start of the full qualified path ('{qualified_class_name}')."
         )

